source/framework/core/src/TRestDataSet.cxx

```diff
@@ -0,0 +1,543 @@
+/// selected by the user, and the corresponding dataset file selection.
+/// The root file will also contain a TRestDataSet object to allow future
+/// users of the output file generated to identify the origin of the data.
+///
+/// - `txt` or `csv`: It will create an ASCII table with a table where each
+/// column will contain the data of a given branch. A header will be
+/// written inside the file with all the information found inside the
+/// TRestDataSet instance.
+///
+///
+/// Example:
+/// \code
+/// restRoot
+/// [0] TRestDataSet d("dataset");
+/// [1] d.Initialize();
+/// [2] d.Export("mydataset.csv");
+/// [3] d.Export("mydataset.root");
+/// \endcode
+///
+///----------------------------------------------------------------------
+///
+/// REST-for-Physics - Software for Rare Event Searches Toolkit
+///
+/// History of developments:
+///
+/// 2022-November: First implementation of TRestDataSet
+/// Javier Galan
+///
+/// \class TRestDataSet
+/// \author: Javier Galan (javier.galan.lacarra@cern.ch)
+///
+/// <hr>
+///
+#include "TRestDataSet.h"
+
+#include "TRestRun.h"
+#include "TRestTools.h"
+
+ClassImp(TRestDataSet);
+
+///////////////////////////////////////////////
+/// \brief Default constructor
+///
+TRestDataSet::TRestDataSet() { Initialize(); }
+
+/////////////////////////////////////////////
+/// \brief Constructor loading data from a config file
+///
+/// If no configuration path is defined using TRestMetadata::SetConfigFilePath
+/// the path to the config file must be specified using full path, absolute or
+/// relative.
+///
+/// The default behaviour is that the config file must be specified with
+/// full path, absolute or relative.
+///
+/// \param cfgFileName A const char* giving the path to an RML file.
+/// \param name The name of the specific metadata. It will be used to find the
+/// corresponding TRestAxionMagneticField section inside the RML.
+///
+TRestDataSet::TRestDataSet(const char* cfgFileName, const std::string& name) : TRestMetadata(cfgFileName) {
+    LoadConfigFromFile(fConfigFileName, name);
+
+    if (GetVerboseLevel() >= TRestStringOutput::REST_Verbose_Level::REST_Info) PrintMetadata();
+}
+
+///////////////////////////////////////////////
+/// \brief Default destructor
+///
+TRestDataSet::~TRestDataSet() {}
+
+///////////////////////////////////////////////
+/// \brief It will initialize the data frame with the filelist and column names
+/// (or observables) that have been defined by the user.
+///
+void TRestDataSet::Initialize() {
+    SetSectionName(this->ClassName());
+
+    if (fTree != nullptr) {
+        RESTWarning << "Tree has already been loaded. Skipping TRestDataSet::LoadDataFrame ... " << RESTendl;
+        return;
+    }
+
+    if (fFileSelection.empty()) FileSelection();
+
+    // We are not ready yet
+    if (fFileSelection.empty()) return;
+
+    ///// Disentangling process observables --> producing finalList
+    TRestRun* run = new TRestRun(fFileSelection[0]);
```

Comment:
I think there is no reason to use a pointer here, we should avoid use `new` and `delete`